### PR TITLE
Fix several bugs in item grant advancement

### PR DESCRIPTION
**Pull request by arbron**
- Fix issue with item grant choices not appearing in advancement list or advancement config due to `TextEditor._createContentLink` returning node rather than raw HTML
- Fix issue with advancement values not being unset when leveling down due to `mergeObject` no long performing deletions by default
- Fix issue with granted item ID not being stored in advancement data

### CODE REVIEW

File: module/advancement/advancement.mjs [hunk 1]
```diff
@@ -259,8 +259,8 @@ export default class Advancement {
     const advancement = foundry.utils.deepClone(this.item.system.advancement);
     const idx = advancement.findIndex(a => a._id === this.id);
     if ( idx < 0 ) throw new Error(`Advancement of ID ${this.id} could not be found to update`);
-    foundry.utils.mergeObject(this.data, updates);
-    foundry.utils.mergeObject(advancement[idx], updates);
+    foundry.utils.mergeObject(this.data, updates, { performDeletions: true });
+    foundry.utils.mergeObject(advancement[idx], updates, { performDeletions: true });
     this.item.updateSource({"system.advancement": advancement});
```
Comment (aaclayton) [inline module/advancement/advancement.mjs:264]: It *should* now in V10 be possible to do this without merging first:
```js
this.item.updateSource({[`system.advancement.${idx}`]: updates});
```

File: module/advancement/types/item-grant.mjs [hunk 2]
```diff
@@ -71,19 +71,17 @@ export class ItemGrantAdvancement extends Advancement {
     for ( const [uuid, selected] of Object.entries(data) ) {
       if ( !selected ) continue;
 
-      const ri = retainedData[uuid];
-      if ( ri && !ri._id ) ri._id = foundry.utils.randomID();
-      const item = ri ? new Item.implementation(ri) : (await fromUuid(uuid))?.clone();
-      if ( !item ) continue;
-
-      item.updateSource({
+      const itemData = retainedData[uuid] ?? (await fromUuid(uuid))?.clone().toObject();
```
Comment (aaclayton) [inline module/advancement/types/item-grant.mjs:74]: You can clean this up a bit by passing in the extra data as part of the `clone` operation:
```js
let itemData = retainedData[uuid];
if ( !itemData ) {
  const source = await fromUuid(uuid);
  itemData = source.clone({
    _id: foundry.utils.randomID(),
    "flags.dnd5e.sourceId": uuid,
    "flags.dnd5e.advancementOrigin": `${this.item.id}.${this.id}`
  }, {keepId: true}).toObject();
}
```

**Review by aaclayton — CHANGES_REQUESTED** [review]
Two comments for you to investigate.

File: module/advancement/advancement.mjs (hunk 1 above)
Comment (arbron) [reply module/advancement/advancement.mjs:264]: That seems to be saving it as an array, not an object. Does this only work with `DataModel` enforced types?

Comment (aaclayton) [reply module/advancement/advancement.mjs:264]: > That seems to be saving it as an object, not an array. Does this only work with `DataModel` enforced types?

Ah, yes. My mistake. It only works when the data model specifically defines an `ArrayField` type. Sorry, maybe one for the future!

**Review by aaclayton — APPROVED** [review]

**Pull request merged**